docs/samples/Microsoft.ML.Samples/Dynamic/TensorFlow/ImageClassification.cs

```diff
@@ -111,11 +115,18 @@ class OutputScores
             public float[] output { get; set; }
         }
 
-        private static string Download(string baseGitPath, string dataFile)
+        private static async Task<string> Download(string baseGitPath, string dataFile)
         {
-            using (WebClient client = new WebClient())
+            if (File.Exists(dataFile))
+                return dataFile;
+
+            using (HttpClient client = new HttpClient())
             {
-                client.DownloadFile(new Uri($"{baseGitPath}"), dataFile);
+                var response = await client.GetStreamAsync(new Uri($"{baseGitPath}")).ConfigureAwait(false);
+                using (var fs = new FileStream(dataFile, FileMode.CreateNew))
+                {
+                    await response.CopyToAsync(fs).ConfigureAwait(false);
```

Comment:
Do you have any guidance on ConfigureAwait(false) use in framework?

In VS world we have a bunch of rules 


- https://github.com/microsoft/vs-threading/blob/main/doc/threading_rules.md
- https://devblogs.microsoft.com/premier-developer/asynchronous-and-multithreaded-programming-within-vs-using-the-joinabletaskfactory/

Reply:
@ericstj any thoughts on this?

Reply:
it should consistently be `ConfigureAwait(false)` in library code because you don't know the synchonization model of whatever app model you are loaded into, so there is no reason to limit your performance.

https://devblogs.microsoft.com/dotnet/configureawait-faq/

This is consistently done in the core libraries.

Reply:
Thanks @danmoseley!